predicators/perception/spot_perceiver.py

```diff
@@ -618,21 +619,24 @@ def __init__(self) -> None:
         self._curr_env: Optional[BaseEnv] = None
         self._waiting_for_observation = True
         self._ordered_objects: List[Object] = []  # list of all known objects
+        self._state_history: Deque[State] = deque(
```

Comment:
Use a list for simplicity for both of these histories -- appending to the end is O(1) in both cases of List vs. Deque